### DISCUSSION TITLE
Can "multi-segment language detection" detect multiple languages in an audio

### DISCUSSION THREAD

**Discussion by toanhuynhnguyen** [SYSTRAN/faster-whisper#1026] [Q&A]
Can "multi-segment language detection" detect multiple languages in an audio?
Thanks.

**Comment by Hikari-Tsai** [comment]
It can transcribe code-switched audio by enabling per-segment language detection:

```python
segments, info = model.transcribe(
    "mixed-language.wav",
    multilingual=True,
    word_timestamps=True,
)
```

Without `multilingual=True`, faster-whisper detects one language near the start and uses that tokenizer language for the whole transcription. With it enabled, detection runs again for each model window and the tokenizer can change languages.

There are important limitations:

- a model window is much larger than an individual word or sentence, so rapid code-switching may still be decoded under one language token;
- `TranscriptionInfo.language` remains the initial/global language;
- the public `Segment` result currently does not include the detected language for each window.

Therefore this option helps multilingual decoding but is not a general language-diarization API. If you need exact language intervals, split the audio into speech regions and run a dedicated language-ID step per region before transcription.
